src/oops/SOLID/lsp/stack/before/Stack.java

```diff
@@ -14,29 +16,33 @@
  * so objects of ArrayList are not fully replaceable by the objects of stack.   
  * 
  */
-public class StackWrong extends ArrayList<Integer>{
+public class Stack {
 	private int topPointer = 0;
-	
+	private List<Integer> list;
+
+	public Stack() {
+		this.list = new ArrayList<>();
+	}
+
+
 	public void push(Integer a) {
-		add(topPointer, a);
+		list.add(topPointer, a);
 		topPointer++;
 	}	
 	public void pop() {
-		remove(topPointer-1);
+		list.remove(topPointer-1);
 		topPointer--;
 	}
 	public Integer top() {
-		return get(topPointer-1);
+		return list.get(topPointer-1);
```

Comment:
Possibility of ArrayIndexOutOfBound Exception here, but you removed the extra top() call in you main() so no issues.

Reply:
Nice catch. Thanks for mentioning.